src/scripts/browserDeviceProfile.js

```diff
@@ -944,21 +944,24 @@ import browser from './browser';
         // Subtitle profiles
         // External vtt or burn in
         profile.SubtitleProfiles = [];
-        if (supportsTextTracks()) {
-            profile.SubtitleProfiles.push({
-                Format: 'vtt',
-                Method: 'External'
-            });
-        }
-        if (options.enableSsaRender !== false && (!options.isRetry && appSettings.get('subtitleburnin') !== 'allcomplexformats')) {
-            profile.SubtitleProfiles.push({
-                Format: 'ass',
-                Method: 'External'
-            });
-            profile.SubtitleProfiles.push({
-                Format: 'ssa',
-                Method: 'External'
-            });
+        const subtitleBurninSetting = appSettings.get('subtitleburnin');
+        if (!options.isRetry && subtitleBurninSetting !== 'all') {
+            if (supportsTextTracks()) {
+                profile.SubtitleProfiles.push({
+                    Format: 'vtt',
+                    Method: 'External'
+                });
+            }
+            if (options.enableSsaRender !== false && subtitleBurninSetting !== 'allcomplexformats') {
+                profile.SubtitleProfiles.push({
+                    Format: 'ass',
+                    Method: 'External'
+                });
+                profile.SubtitleProfiles.push({
+                    Format: 'ssa',
+                    Method: 'External'
+                });
+            }
```

Comment:
_from Matrix_
To respect the old logic:
- burn in text subtitles only by user's decision.
- burn in SSA/ASS if playback error occurs or by user's decision.
`SubtitlesOctopus` can throw `mediadecodeerror`. _It might be better to have a different error._
https://github.com/jellyfin/jellyfin-web/blob/d96b6ac4ee6899e3d389250eee10de2634785975/src/plugins/htmlVideoPlayer/plugin.js#L1067-L1069

```suggestion
        const subtitleBurninSetting = appSettings.get('subtitleburnin');
        if (subtitleBurninSetting !== 'all') {
            if (supportsTextTracks()) {
                profile.SubtitleProfiles.push({
                    Format: 'vtt',
                    Method: 'External'
                });
            }
            if (options.enableSsaRender !== false && !options.isRetry && subtitleBurninSetting !== 'allcomplexformats') {
                profile.SubtitleProfiles.push({
                    Format: 'ass',
                    Method: 'External'
                });
                profile.SubtitleProfiles.push({
                    Format: 'ssa',
                    Method: 'External'
                });
            }
```